novawallet/Common/Services/Web3AlertService/Web3AlertsSyncService.swift

```diff
@@ -0,0 +1,174 @@
+import RobinHood
+import SoraFoundation
+import SoraKeystore
+import FirebaseCore
+import FirebaseFirestore
+import FirebaseMessaging
+
+enum Web3AlertsSyncServiceError: Error {
+    case documentNotFound
+}
+
+protocol Web3AlertsSyncServiceProtocol: ApplicationServiceProtocol {
+    func save(settings: LocalPushSettings) -> BaseOperation<Void>
+    func update(token: String) -> CompoundOperationWrapper<Void?>
```

Comment:
Interface probably must accept completion closure instead of returning wrappers. Currently the problem is that nobody tracking currently executing operation. As we discussed we don't want to execute several operations at the same time to the remote. Thus we need to save locally current executing wrapper, clear it when it is completed don't sync if we already executing something and addDependency to it if we updating/saving.